### DISCUSSION TITLE
JavaScript SDK - enable SSO / OAuth

### DISCUSSION THREAD

**Discussion by habtim** [directus/directus#11156] [Archive]
## Scenario
We have a SPA with SSO Azure Authentication. Our application should consume the directus api for extra content, preferably by the JavaScript SDK. 
Currently, this is only achievable by using basic auth (user & pw) or static token, both beeing insecure. 

## What we tried so far:
1. Use our access_token from the application with `Auth.updateStorage(authResult)` ("private" method) to fake a login. But we get a Azure specific error `90009` - Application is requesting a token for itself. I guess the api wants a separate token with a separate client_id in the background.
2. [Docs](https://docs.directus.io/reference/authentication/#login-using-sso-providers) is not an option, as we have to route away from the SPA, which will not route back to our app again.

## Possible solution
Provide an external SSO or/and OAuth (Code Flow + PKCE) possiblity for the jsSDK.

**Comment by joselcvarela** [comment]
Thank you for taking your time to suggest this feature.
But there's already an open discussion regarding this https://github.com/directus/directus/discussions/4463
You may want to go there and upvote 👍
